ci/scripts/cpp_build.sh

```diff
@@ -31,11 +31,19 @@ elif [ -x "$(command -v xcrun)" ]; then
   export ARROW_GANDIVA_PC_CXX_FLAGS="-isysroot;$(xcrun --show-sdk-path)"
 fi
 
-# export CCACHE_COMPRESS=1
-# export CCACHE_COMPRESSLEVEL=5
-# export CCACHE_COMPILERCHECK=content
-# export CCACHE_DIR=/build/ccache
-# export PATH=/usr/lib/ccache/:$PATH
+export ARROW_USE_CCACHE=${ARROW_USE_CCACHE:-ON}
+
+if [ "${ARROW_USE_CCACHE}" == "ON" ]; then
+    export CCACHE_COMPILERCHECK=content
+    export CCACHE_COMPRESS=1
+    export CCACHE_COMPRESSLEVEL=5
+    export CCACHE_DIR=/build/ccache
```

Comment:
What if this directory doesn't exists and the user doesn't set ARROW_USE_CCACHE=OFF explicitly?
Perhaps it would be better to set CCACHE_DIR from outside and use OFF for ARROW_USE_CCACHE by default, then set it to ON in the base images.

Reply:
Well, the directory doesn't exist by default. AFAICT, it's created by ccache automatically.

That said, the proposed changes sound good to me :-)

Reply:
Hmm, is there a way to set those environment variables by default for all docker-compose entries and/or Dockerfiles? Otherwise it's gonna be annoying to do it one by one.

Reply:
It is indeed frustrating. I propose another workaround, set `CCACHE_DIR="${build_dir}/ccache"`. The `build_dir` is received as an argument, so it can be altered from outside of the scripts.

Reply:
Hmm, that's not good. The build dir is a CMake build directory, and we don't want to wipe the ccache if we remove the build directory.

Reply:
We can use `.env` to define default environment values.

Reply:
We don't have any `ARROW_xxx` variable in `.env` currently.